src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,27 @@
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { queryChange, statusChange } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { query, status } = useAppSelector(state => state.filter);
+
+  const handleQueryChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(queryChange(e.target.value));
+  };
+
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select
+            data-cy="statusSelect"
+            value={status}
+            onChange={e => dispatch(statusChange(e.target.value))}
```

Comment:
Consider adding a type for the event parameter in the onChange handler to ensure type safety. For example, you can use (e: React.ChangeEvent<HTMLSelectElement>) => dispatch(statusChange(e.target.value)).